decentraland/screens/helpers.js

```diff
@@ -1,36 +1,175 @@
+  const value = 1e18; // one
+  const contracts = getContracts();
+  const { manaContract, marketplaceContract } = contracts;
+  const toAddress = marketplaceContract.getAddress();
+  manaContract.setWallet(fromAccount);
+  const action = manaContract.approve(toAddress, `${value}`);
+  await action.waitForNonceToUpdate();
+};
+
+export const addressesAreEqual = (address1, address2) => {
+  return address1.toUpperCase() === address2.toUpperCase();
+};
+
+export const prepareEstateForSale = async (estateContract, estateForSale) => {
+  const { id, assetId, seller, priceInWei, expiresAt } = estateForSale;
+
+  const estateId = Number(assetId);
+
+  // Note: Conversion to USD will be implemented on v0.2.0
+  const manaPerUsd = 30;
+  const priceMana = Number(`${priceInWei}`) / 1e18;
+  const priceUSD = Number(priceMana / manaPerUsd).toFixed(2);
+  const name = await estateContract.getMetadata(assetId);
+  const imgUrl = `https://api.decentraland.org/v1/estates/${estateId}/map.png`;
+
+  return {
+    id,
+    priceMana,
+    priceUSD,
+    seller,
+    expiresAt,
+    type: ESTATE,
+    asset: {
+      id: estateId,
+      name,
+      img: imgUrl,
+    },
+  };
 };
 
-export default { approveManaSpending, manaFaucetTo };
+export const prepareParcelForSale = async (landContract, parcelForSale) => {
+  const {
+    id,
+    assetId: parcelId,
+    seller,
+    priceInWei,
+    expiresAt,
+  } = parcelForSale;
+
+  // Note: Conversion to USD will be implemented on v0.2.0
+  const manaPerUsd = 30;
+  const priceMana = Number(`${priceInWei}`) / 1e18;
+  const priceUSD = Number(priceMana / manaPerUsd).toFixed(2);
+  const namePromise = landContract.tokenMetadata(parcelId);
+  const coordsPromise = landContract.decodeTokenId(parcelId);
+  const [name, coords] = await Promise.all([namePromise, coordsPromise]);
+  const [x, y] = coords;
+  const imgUrl = `https://api.decentraland.org/v1/parcels/${x}/${y}/map.png`;
+
+  return {
+    id,
+    priceMana,
+    priceUSD,
+    seller,
+    expiresAt,
+    type: PARCEL,
+    asset: {
+      id: parcelId,
+      name,
+      img: imgUrl,
+    },
+  };
+};
+
+export const fundAccount = async accountAddress => {
+  const SMALL_AMOUNT = `${1e17}`; // 0.1
+  const TEN = `${10e18}`;
+
+  const gnosisSafeOwnerPrivKey =
+    "0xee0c6b1a7adea9f87b1a422eb06b245fc714b8eca4c8c0578d6cf946beba86f1";
+  const gnosisSafeOwner = createFromPrivateKey(gnosisSafeOwnerPrivKey);
+
+  const contracts = getContracts();
+  const { manaContract, gnosisSafeContract } = contracts;
+  gnosisSafeContract.setWallet(gnosisSafeOwner);
+  gnosisSafeContract.setSigners([gnosisSafeOwner]);
+
+  const transferEthersAction = gnosisSafeContract.transferEther(
+    accountAddress,
+    SMALL_AMOUNT
+  );
+  await transferEthersAction.waitForNonceToUpdate();
+
+  const transferManaAction = gnosisSafeContract.transferERC20(
+    manaContract.getAddress(),
+    accountAddress,
+    TEN
+  );
+  await transferManaAction.waitForNonceToUpdate();
```

Comment:
Should we have a try/catch here?

Reply:
The `waitForNonceToUpdate` function won't throw errors, we'll be better listen to error events from contracts and actions.
I've tried to make that changes and strip off all `waitForNonceToUpdate` but since the app becomes slower after that, I've back to the way that it's and stay focused on our short term goal. I've just created the issue https://github.com/tasitlabs/tasit/issues/151 to keep that in mind.

Reply:
Yep. It it worth creating a separate issue that we should respond to more errors like reverts even if that's not exclusively via events?

Reply:
That makes sense and sounds like the right move - thanks!

Reply:
Feel free to resolve this thread after reading

Reply:
I've made a note about that on the issue https://github.com/tasitlabs/tasit/issues/159.
Resolving this.